From eeb78cad48b50da7150d53ee7f22a95d79f9eb13 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Tue, 16 Sep 2014 17:04:04 -0400
Subject: [usb] serial: fix write memory-allocation flag

Message-id: <1410887463-20674-173-git-send-email-dzickus@redhat.com>
Patchwork-id: 95237
O-Subject: [RHEL7 PATCH 172/591] USB: serial: fix write memory-allocation flag
Bugzilla: 1110939
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1110939

commit 043e3f834530e15e89c58c1b7af59cc646700134
Author: Johan Hovold <jhovold@gmail.com>
Date:   Sat Nov 9 12:38:10 2013 +0100

    USB: serial: fix write memory-allocation flag

    Fix regression introduced by commit 818f60365a29 ("USB: serial: add
    memory flags to usb_serial_generic_write_start"), which incorrectly used
    GFP_KERNEL in write(), which must not not sleep.

    Reported-by: Dave Jones <davej@fedoraproject.org>
    Tested-by: Dave Jones <davej@fedoraproject.org>
    Cc: Dave Jones <davej@fedoraproject.org>
    Signed-off-by: Johan Hovold <jhovold@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/usb/serial/generic.c b/drivers/usb/serial/generic.c
index e36b25a..b63ce02 100644
--- a/drivers/usb/serial/generic.c
+++ b/drivers/usb/serial/generic.c
@@ -200,7 +200,7 @@ int usb_serial_generic_write(struct tty_struct *tty,
   return 0;
 
  count = kfifo_in_locked(&port->write_fifo, buf, count, &port->lock);
- result = usb_serial_generic_write_start(port, GFP_KERNEL);
+ result = usb_serial_generic_write_start(port, GFP_ATOMIC);
  if (result)
   return result;
 
-- 
1.7.1